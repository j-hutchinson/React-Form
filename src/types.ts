
export enum FormState {
    USER = 'User',
    PRIVACY = 'Privacy',
    DONE = 'Done',
}

export interface ReduxState {
    user: UserSettings;
    privacy: PrivacySettings;
}

export interface UserSettings {
    name: string;
    title: string;
    email: string;
    password: string
}

export interface PrivacySettings {
    updates: boolean;
    communication: boolean;
}

export interface ActionType {
    type: string;
    privacy?: PrivacySettings;
    user?: UserSettings;
}