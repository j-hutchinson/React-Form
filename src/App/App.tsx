import React, { useState } from 'react';
import styled from 'styled-components';

import FormBody from '../FormBody/component';
import FormTabs from '../FormTabs/component';
import { FormState } from '../types'

const StyledApp = styled.div`
   display: flex;
   margin-top: 100px;
`;

const StyledForm = styled.div`
   display: flex;
   flex-direction: column;
   width: 600px;
   margin: auto;
`;

const App = (): JSX.Element => {
    const [activeStep, setActiveStep] = useState<FormState>(FormState.USER);

    return (
        <StyledApp>
            <StyledForm>
                <FormTabs activeStep={activeStep} />
                <FormBody
                    activeStep={activeStep}
                    setActiveStep={setActiveStep}
                />
            </StyledForm>
        </StyledApp>
    );
};

export default App;
