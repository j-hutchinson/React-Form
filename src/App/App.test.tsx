import { shallow } from 'enzyme';
import React from 'react';
import { App } from './App';
import { FormState } from '../types';

describe('App component', () => {
    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<App activePage={FormState.User} />);

        expect(wrapper).toMatchSnapshot();
    });
});
