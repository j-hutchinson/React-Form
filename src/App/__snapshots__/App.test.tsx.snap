// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <FormTabs
      activeStep="User"
    />
    <FormBody
      activeStep="User"
      setActiveStep={[Function]}
    />
  </styled.div>
</styled.div>
`;
